src/qibolab/instruments/dummy.py

```diff
@@ -73,7 +76,10 @@ def sweep(
 
         for ro_pulse in sequence.ro_pulses:
             if options.acquisition_type is AcquisitionType.DISCRIMINATION:
-                values = np.random.rand(expts)
+                if options.averaging_mode is AveragingMode.SINGLESHOT:
+                    values = np.random.randint(2, size=expts)
+                elif options.averaging_mode is AveragingMode.CYCLIC:
+                    values = np.random.rand(expts)
```

Comment:
It's twice the same, so it is worth to make a tiny function for this :)

Reply:
I think the whole loops for play and sweep are, I will create a function for both